pkg/controller/controller_instance_test.go

```diff
@@ -215,7 +215,63 @@ func TestReconcileInstanceNonExistentServicePlan(t *testing.T) {
 	}
 }
 
-func TestReconcileInstanceWithParameters(t *testing.T) {
+func TestReconcileInstanceWithInvalidParameters(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestInstance()
+
+	// Secret with invalid JSON
+	invalidSecret := &v1.Secret{
+		Data: map[string][]byte{
+			testSecretKey: []byte("invalid json"),
```

Comment:
@duglin yeah, I see what you mean, and I tend to agree that we should treat values as just arbitrary strings.

> Can we determine if a value is a string, boolean, or number if it isn't encoded? Is secret value true the boolean true or string "true"?

@spadgett  I don't think we can and should. Only if we apply a priority list of "duck typing" rules (and I don't like this idea, just giving a **bad** example of supporting this):
- if the string contains valid JSON, treat it as a `JSON tree`
- if the string is equal to "true" or "false", it is `boolean`
- if the string contains only digits, it is `int`
- ...
- if there is no match then, well... string is a string

Again, I think it is a bad idea. What if the user wanted to pass "true" as a string?
I would prefer to treat the value always the same way. And to make it easier for the users to use, I propose to treat multi-key secrets as Map<String, String> (without any magic duck typing). Any other use case is solved by single-key JSON blob, where you can put whatever you want, and we will pass it "as is".